### ISSUE TITLE
generate-list-of-rules:check is failing on main

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [jsx-eslint/eslint-plugin-jsx-a11y#912]
```shell
git checkout main
git pull
npm i
npm run generate-list-of-rules:check
```

```plaintext
> eslint-plugin-jsx-a11y@6.6.1 generate-list-of-rules:check
> npm run generate-list-of-rules -- --check


> eslint-plugin-jsx-a11y@6.6.1 pregenerate-list-of-rules
> npm run build


> eslint-plugin-jsx-a11y@6.6.1 build
> rimraf lib && babel src --out-dir lib --copy-files

Successfully compiled 101 files with Babel (579ms).

> eslint-plugin-jsx-a11y@6.6.1 generate-list-of-rules
> eslint-doc-generator --rule-doc-title-format prefix-name --rule-doc-section-options false --config-emoji recommended,☑️

Please run eslint-doc-generator. A rule doc is out-of-date: docs/rules/label-has-for.md
- Expected
+ Received

@@ -1,8 +1,8 @@
  # jsx-a11y/label-has-for

- ❌ This rule is deprecated. It was replaced by [`jsx-a11y/label-has-associated-control`](label-has-associated-control.md).
+ ❌ This rule is deprecated. It was replaced by [`jsx-a11y/label-has-associated-control`](../../docs/rules/label-has-associated-control.md).

  🚫 This rule is _disabled_ in the following configs: ☑️ `recommended`, 🔒 `strict`.

  <!-- end auto-generated rule header -->
```

Seen in the failing action run on #835: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/actions/runs/3875020488/jobs/6606979150.
